src/LiveComponent/src/EventListener/AddLiveAttributesSubscriber.php

```diff
@@ -73,6 +77,17 @@ public function onPreRender(PreRenderEvent $event): void
                     $originalAttributes['data-embedded-template-index'],
                 );
             }
+
+            $inputProps = $event->getMountedComponent()->getInputProps();
+            $isDeferred = array_key_exists('defer', $inputProps);
+
+            if ($isDeferred) {
+                $event->setTemplate('@LiveComponent/deferred.html.twig');
+
+                $variables['loadingTag'] = $originalAttributes['loading-tag'] ?? self::DEFAULT_LOADING_TAG;
+                $variables['loadingTemplate'] = $originalAttributes['loading-template'] ?? self::DEFAULT_LOADING_TEMPLATE;
+                $attributes = $attributes->without('defer', 'loading-element', 'loading-template');
+            }
```

Comment:
I still have a preference to have this in its own event listener.

Reply:
You want it, you have it 🫡!